### PR TITLE
bug fix libmagic optional (#17)

### PR DESCRIPTION
**Pull request by dinhlongviolin1**
Solve Bug #17
- Make libmagic and rdp optional
- use util.find_spec in tests so tests work properly
- update setup.py
- tests with optional dependencies
- all tests suites now run in both ubuntu and windows with python 3.8, 3.9, 3.10

### CODE REVIEW

File: taipy/gui/data/utils.py [hunk 1]
```diff
@@ -16,6 +24,8 @@ def df_data_filter(
     expected_row_count: t.Union[None, int],
     margin_of_error: float = 0.3,
 ) -> t.Tuple[pd.DataFrame, t.Union[None, str], str]:
+    if not _has_rdp_module:
+        return None
```
Comment (FabienLelaquais) [inline taipy/gui/data/utils.py]: I guess, if RDP is not there, that I would return the original dataframe...
Am I missing anything?

**Review by FabienLelaquais — APPROVED** [review]
Looking good except for this little question of mine...

File: taipy/gui/data/utils.py (hunk 1 above)
Comment (dinhlongviolin1) [reply taipy/gui/data/utils.py]: Thanks for this, I returned the wrong thing :D

Comment (dinhlongviolin1) [reply taipy/gui/data/utils.py]: Updated

File: tests/taipy/gui/control/test_image.py [hunk 2]
```diff
@@ -22,6 +24,9 @@ def test_image_file_md(gui: Gui, helpers):
             "<Image",
             'defaultContent="data:image/png;base64,',
         ]
+        if not util.find_spec("magic"):
+            expected_list = ["<Image", 'defaultContent="/taipy-content/taipyStatic0/TaiPyContent.', ".bin"]
+        print(expected_list)
```
Comment (FredLL-Avaiga) [inline tests/taipy/gui/control/test_image.py]: Remove this ?

**Review by FredLL-Avaiga — APPROVED** [review]

File: tests/taipy/gui/control/test_image.py (hunk 2 above)
Comment (dinhlongviolin1) [reply tests/taipy/gui/control/test_image.py]: thanks

**Pull request merged**